tests/workload/transactions.go

```diff
@@ -107,42 +106,58 @@ func GenerateNBlocks(ctx context.Context, require *require.Assertions, uris []st
 	}
 }
 
-func GenerateUntilCancel(
-	ctx context.Context,
+func GenerateUntilStop(
+	require *require.Assertions,
 	uris []string,
 	generator TxWorkloadIterator,
+	stopChannel <-chan struct{},
 ) {
 	submitClient := jsonrpc.NewJSONRPCClient(uris[0])
+	ctx := context.Background()
```

Comment:
Could we add `ctx` to this function signature and keep passing it in as before? We should just change from cancelling at the end of the function to terminating via the channel, so that we don't need to handle ctx errors